app/controllers/conseiller/experts_controller.rb

```diff
@@ -27,7 +27,9 @@ def retrieve_experts
     expert_with_code_size = @experts.size
 
     if expert_with_code_size < 10
+      primary_ids = @experts.pluck(:id)
       additional_experts = base_query
+        .where.not(id: primary_ids)
```

Comment:
Du coup, autant écrire ici:
```suggestion
        .where.not(id: @experts.ids)
```

En fait, en principe on pourrait écrire directement 
```suggestion
        .where.not(id: @experts)
```
et laisser ActiveRecord construire une subquery si nécessaire, mais ici, ce n’est pas pertinent, parce que:
- `@experts` devra être chargée pour elle-même, pas uniquement comme subquery
- Ce n’est pas possible simplmeent, parce qu’on a ajouté une colonne `source` au select, et que ActiveRecord génère un SQL invalide (je n’ai pas fouillé pourquoi).

(Ce que je voulais dire en substance c’est que: en général, il faut éviter de charger les .ids soi-même et laisser ActiveRecord générer des subqueries. Mais pas ici. Bref.)